File: medplum-java/medplum-model/src/main/java/com/fhir/SearchParameter.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.Boolean;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutableSearchParameter.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface SearchParameter {
  Optional<canonical> derivedFrom();

  Optional<List<UsageContext>> useContext();

  Optional<SearchparameterStatus> status();

  Optional<dateTime> date();

  Optional<Narrative> text();

  Optional<SearchparameterType> type();

  Optional<List<CodeableConcept>> jurisdiction();

  Optional<markdown> description();

  Optional<List<SearchparameterModifierItem>> modifier();

  Optional<Meta> meta();

  Optional<SearchparameterXpathusage> xpathUsage();

  Optional<List<Extension>> modifierExtension();

  String resourceType();

  Optional<uri> url();

  Optional<String> version();

  Optional<String> xpath();

  Optional<List<SearchparameterComparatorItem>> comparator();

  Optional<String> name();

  Optional<List<code>> target();

  Optional<Boolean> experimental();

  Optional<code> code();

  Optional<List<ContactDetail>> contact();

  Optional<id> id();

  Optional<Boolean> multipleAnd();

  Optional<List<Extension>> extension();

  Optional<String> publisher();

  Optional<List<ResourceList>> contained();

  Optional<code> language();

  Optional<markdown> purpose();

  Optional<List<code>> base();

  Optional<Boolean> multipleOr();

  Optional<List<String>> chain();

  Optional<uri> implicitRules();

  Optional<String> expression();

  Optional<List<SearchParameter_Component>> component();

  static ImmutableSearchParameter.ResourceTypeBuildStage builder() {
    return ImmutableSearchParameter.builder();
  }
}

File: medplum-typescript/src/model/fhir/SearchParameter.ts
import { canonical } from "./canonical";
import { code } from "./code";
import { CodeableConcept } from "./CodeableConcept";
import { ContactDetail } from "./ContactDetail";
import { dateTime } from "./dateTime";
import { Extension } from "./Extension";
import { id } from "./id";
import { markdown } from "./markdown";
import { Meta } from "./Meta";
import { Narrative } from "./Narrative";
import { ResourceList } from "./ResourceList";
import { SearchparameterComparatorItem } from "./SearchparameterComparatorItem";
import { SearchparameterModifierItem } from "./SearchparameterModifierItem";
import { SearchparameterStatus } from "./SearchparameterStatus";
import { SearchparameterType } from "./SearchparameterType";
import { SearchparameterXpathusage } from "./SearchparameterXpathusage";
import { SearchParameter_Component } from "./SearchParameter_Component";
import { uri } from "./uri";
import { UsageContext } from "./UsageContext";

export interface SearchParameter {
    resourceType: string;
    id: id | null | undefined;
    meta: Meta | null | undefined;
    implicitRules: uri | null | undefined;
    language: code | null | undefined;
    text: Narrative | null | undefined;
    contained: ResourceList[] | null | undefined;
    extension: Extension[] | null | undefined;
    modifierExtension: Extension[] | null | undefined;
    url: uri | null | undefined;
    version: string | null | undefined;
    name: string | null | undefined;
    derivedFrom: canonical | null | undefined;
    status: SearchparameterStatus | null | undefined;
    experimental: boolean | null | undefined;
    date: dateTime | null | undefined;
    publisher: string | null | undefined;
    contact: ContactDetail[] | null | undefined;
    description: markdown | null | undefined;
    useContext: UsageContext[] | null | undefined;
    jurisdiction: CodeableConcept[] | null | undefined;
    purpose: markdown | null | undefined;
    code: code | null | undefined;
    base: code[] | null | undefined;
    type: SearchparameterType | null | undefined;
    expression: string | null | undefined;
    xpath: string | null | undefined;
    xpathUsage: SearchparameterXpathusage | null | undefined;
    target: code[] | null | undefined;
    multipleOr: boolean | null | undefined;
    multipleAnd: boolean | null | undefined;
    comparator: SearchparameterComparatorItem[] | null | undefined;
    modifier: SearchparameterModifierItem[] | null | undefined;
    chain: string[] | null | undefined;
    component: SearchParameter_Component[] | null | undefined;
}
